src/ir.h

```diff
@@ -101,40 +114,71 @@ struct Const {
   uintptr_t ref_bits() const { return data_.To<uintptr_t>(0); }
   v128 vec128() const { return data_; }
 
-  bool is_expected_nan() const { return nan_ != ExpectedNan::None; }
-  ExpectedNan expected() const { assert(is_expected_nan()); return nan_; }
+  template <typename T>
+  T v128_lane(int lane) const { return data_.To<T>(lane); }
 
   void set_u32(uint32_t x) { From(Type::I32, x); }
   void set_u64(uint64_t x) { From(Type::I64, x); }
   void set_f32(uint32_t x) { From(Type::F32, x); }
   void set_f64(uint64_t x) { From(Type::F64, x); }
-  void set_vec128(v128 x)  { From(Type::V128, x); }
+
+  void set_v128_u8(int lane, uint8_t x) { set_v128_lane(lane, Type::I8, x); }
+  void set_v128_u16(int lane, uint16_t x) { set_v128_lane(lane, Type::I16, x); }
+  void set_v128_u32(int lane, uint32_t x) { set_v128_lane(lane, Type::I32, x); }
+  void set_v128_u64(int lane, uint64_t x) { set_v128_lane(lane, Type::I64, x); }
+  void set_v128_f32(int lane, uint32_t x) { set_v128_lane(lane, Type::F32, x); }
+  void set_v128_f64(int lane, uint64_t x) { set_v128_lane(lane, Type::F64, x); }
 
   // Only used for expectations. (e.g. wast assertions)
-  void set_f32(ExpectedNan nan) { From<float>(Type::F32, 0); nan_ = nan; }
-  void set_f64(ExpectedNan nan) { From<double>(Type::F64, 0); nan_ = nan; }
+  void set_f32(ExpectedNan nan) { set_f32(0); set_expected_nan(0, nan); }
+  void set_f64(ExpectedNan nan) { set_f64(0); set_expected_nan(0, nan); }
```

Comment:
Is clang format OK with multiple statements on one line like this?

Reply:
No, it will expand, though I kind of prefer keeping it on a single line for readability. What do you think?

Reply:
Depends what you want the policy for wabt to be ?  Absolute deferral to clang-format, or programmer discretion.    I don't think is one correct answer.   It depends on the project.   We don't have a lot of formatting discussions so allowing discretion isn't currently effecting productivity.  So i'm with with either way right now.   We should probably document it though.

Reply:
Yeah, I'm thinking I'll spend some time writing up a C++ style guide finally this week.